lib/ansible/plugins/action/__init__.py

```diff
@@ -749,6 +749,16 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
 
         self._update_module_args(module_name, module_args, task_vars)
 
+        # TODO: move this code to a separate function
+        # TODO: parse interpreter from module
+        interpreter = 'python'
+        # TODO: make list of binary names to search for configurable on a per-interpreter basis
+        result = self._low_level_execute_command('which python python3', sudoable=False)
+        # The 'which' command will exit with '1' when not all arguments are found, which isn't
+        # necessarily an error.
+        if result['rc'] <= 1 and len(result['stdout_lines']) > 0:
+            task_vars['ansible_' + interpreter + '_interpreter'] = result['stdout_lines'][0]
+
         # FUTURE: refactor this along with module build process to better encapsulate "smart wrapper" functionality
         (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)
```

Comment:
i would execute AFTER getting module's shebang so you can find the 'right interpreter' also check that ansible_x_interpreter does not exist already

Reply:
There's a `TODO` in there already for finding the correct interpreter for the module, and I've already pushed another commit that adds the check for an existing `ansible_x_interpreter` var